npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -38,7 +38,9 @@ def self.yarn_version_numeric(yarn_lock)
 
       sig { params(pnpm_lock: DependencyFile).returns(Integer) }
       def self.pnpm_version_numeric(pnpm_lock)
-        if pnpm_lockfile_version(pnpm_lock).to_f >= 6.0
+        if pnpm_lockfile_version(pnpm_lock).to_f >= 9.0
+          9
+        elsif pnpm_lockfile_version(pnpm_lock).to_f >= 6.0
           8
         elsif pnpm_lockfile_version(pnpm_lock).to_f >= 5.4
           7
```

Comment:
It might be better to invert the checks:
```
let pnpm_version = pnpm_lockfile_version(pnpm_lock).to_f;

if (typeof(pnpm_version) === 'undefined' || pnpm_version  < 5.4)
  6
elsif pnpm_version < 6
  7
elsif pnpm_version < 9
  8
else
  parseInt(pnpm_version, 10)
end
```

Then you'll likely never have to fix it again. ;). Just be done fixing it.